src/spikeinterface/core/waveforms_extractor_backwards_compatibility.py

```diff
@@ -548,6 +552,8 @@ def _read_old_waveforms_extractor_binary(folder, sorting):
         new_params = ext._set_params()
         updated_params = make_ext_params_up_to_date(ext, params, new_params)
         ext.set_params(**updated_params, save=False)
+        if ext.need_backward_compatibility_on_load:
+            ext._handle_backward_compatibility_on_load()
```

Comment:
This must be moved after the `ext.data = ...` because this function can hack the two sub dict (params + data).

So move line 612

Reply:
You mean to move this at the end of the function?

Reply:
I don't think we need to modify the data, since we build it on the fly based on the current API. Or am I getting this wrong?